// ngx-highlightjs/esm2022/plus/highlight-plus.module.mjs
import { NgModule } from '@angular/core';
import { HttpClientModule } from '@angular/common/http';
import { HighlightModule } from 'ngx-highlightjs';
import { GistFilePipe, GistDirective } from './gist';
import { CodeFromUrlPipe } from './code-from-url';
import * as i0 from "@angular/core";
class HighlightPlusModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.2", ngImport: i0, type: HighlightPlusModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.0.2", ngImport: i0, type: HighlightPlusModule, declarations: [GistDirective,
            GistFilePipe,
            CodeFromUrlPipe], imports: [HighlightModule,
            HttpClientModule], exports: [HighlightModule,
            GistDirective,
            GistFilePipe,
            CodeFromUrlPipe] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.0.2", ngImport: i0, type: HighlightPlusModule, imports: [HighlightModule,
            HttpClientModule, HighlightModule] }); }
}
export { HighlightPlusModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.2", ngImport: i0, type: HighlightPlusModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        HighlightModule,
                        HttpClientModule
                    ],
                    declarations: [
                        GistDirective,
                        GistFilePipe,
                        CodeFromUrlPipe
                    ],
                    exports: [
                        HighlightModule,
                        GistDirective,
                        GistFilePipe,
                        CodeFromUrlPipe
                    ]
                }]
        }] });
